src/compositor/input/pointer.rs

```diff
@@ -111,70 +119,37 @@ fn update_view_position(cursor: &mut Cursor, seat: &mut Seat, views: &mut [View]
     }
 }
 
-fn view_at_pointer<'view>(views: &'view mut [View],
-                          cursor: &mut Cursor)
-                          -> (Option<&'view mut View>, Option<SurfaceHandle>, f64, f64) {
+fn view_at_pointer(views: &mut Vec<Rc<View>>,
+                   cursor: &mut Cursor)
+                   -> (Option<Rc<View>>, Option<SurfaceHandle>, f64, f64) {
     for view in views {
-        match view.shell.clone() {
-            Shell::XdgV6(ref mut shell) => {
+        match view.shell {
+            Shell::XdgV6(ref shell) => {
                 let (mut sx, mut sy) = (0.0, 0.0);
-                let surface = with_handles!([(shell: {&mut *shell})] => {
+                let surface = with_handles!([(shell: {shell})] => {
                     let (lx, ly) = cursor.coords();
-                    let Origin {x: shell_x, y: shell_y} = view.origin;
+                    let Origin {x: shell_x, y: shell_y} = view.origin.get();
                     let (view_sx, view_sy) = (lx - shell_x as f64, ly - shell_y as f64);
                     shell.surface_at(view_sx, view_sy, &mut sx, &mut sy)
                 }).unwrap();
                 if surface.is_some() {
-                    return (Some(view), surface, sx, sy)
+                    return (Some(view.clone()), surface, sx, sy)
                 }
             }
         }
     }
     (None, None, 0.0, 0.0)
 }
 
-/// Focus the view under the pointer.
-fn focus_under_pointer<'view, V>(seat: &mut compositor::Seat,
-                                 keyboards: &mut [KeyboardHandle],
-                                 view: V)
-                                 -> HandleResult<()>
-    where V: Into<Option<&'view mut View>>
-{
-    match view.into() {
-        None => {
-            if let Some(mut focused_view) = seat.focused.take() {
-                focused_view.activate(false);
-            }
-            with_handles!([(seat: {&mut seat.seat})] => {
-                seat.keyboard_clear_focus()
-            })
-        }
-        Some(view) => {
-            seat.focused = Some(view.clone());
-            view.activate(true);
-            for keyboard in { &mut *keyboards } {
-                with_handles!([(seat: {&mut seat.seat}),
-                (surface: {view.surface()}),
-                (keyboard: {keyboard})] => {
-                    seat.keyboard_notify_enter(surface,
-                                               &mut keyboard.keycodes(),
-                                               &mut keyboard.get_modifier_masks())
-                })?;
-            }
-            Ok(())
-        }
-    }
-}
-
 /// Start moving a view if you haven't already by passing `start: None`.
 ///
 /// Otherwise, update the view position relative to where the move started,
 /// which is provided by Action::Moving.
-fn move_view<O>(seat: &mut compositor::Seat, cursor: &mut Cursor, view: &mut View, start: O)
+fn move_view<O>(seat: &mut compositor::Seat, cursor: &mut Cursor, view: Rc<View>, start: O)
```

Comment:
If you can use `&View` instead of `Rc<View>` you should (and I believe you can here).

`Rc<T>` derefs into a `&T`